internal/juju/applications.go

```diff
@@ -410,6 +410,13 @@ func (c applicationsClient) deployFromRepository(applicationAPIClient Applicatio
 		return errors.Join(errs...)
 	}
 
+	// publishing to a closed channel automatically upgrade the channel. And this could
+	// lead to an inconsistent terraform state. This check is to error out to the user,
+	// so they can purposely select the track.
+	if deployInfo.Channel != transformedInput.charmChannel {
+		return fmt.Errorf("you are trying to publish to a closed channel. Check the channel for the %s", transformedInput.applicationName)
```

Comment:
in this case the terraform plan is specifying a closed channel, so trying to deploy from a channel, not publish